Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm

```diff
@@ -609,10 +609,14 @@ static WallTime toSystemClockTime(NSDate *date)
 
 void MediaPlayerPrivateAVFoundationObjC::createVideoLayer()
 {
+    assertIsMainThread();
+
     if (!m_avPlayer || m_haveBeenAskedToCreateLayer)
         return;
 
     ensureOnMainThread([this, weakThis = ThreadSafeWeakPtr { *this }] {
```

Comment:
ensureOnMainThread is strange and most probably unneeded given the above assertIsMainThread().

Reply:
probably yes, but this dispatch to the main thread has existed for a while, to scared to remove it :)
Though from looking at the calling point, it is indeed main thread only.

I will remove that assert though, as it's not required at this level.